File: app/src/main/java/me/neko/nzhelper/data/Session.kt
package me.neko.nzhelper.data

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class Session(
    @SerializedName("timestamp") val timestamp: LocalDateTime,
    @SerializedName("duration") val duration: Int,
    @SerializedName("remark") val remark: String,
    @SerializedName("location") val location: String,
    @SerializedName("watchedMovie") val watchedMovie: Boolean,
    @SerializedName("climax") val climax: Boolean,
    @SerializedName("rating") val rating: Float,
    @SerializedName("mood") val mood: String,
    @SerializedName("props") val props: String
)

File: NZhelper/Models/Session.swift
import Foundation

struct Session: Identifiable, Codable, Equatable {
    var id: UUID = UUID()
    var timestamp: Date
    var duration: Int
    var remark: String
    var location: String
    var watchedMovie: Bool
    var climax: Bool
    var rating: Float
    var mood: String
    var props: String

    static var dateFormatter: DateFormatter = {
        let f = DateFormatter()
        f.dateFormat = "yyyy-MM-dd'T'HH:mm:ss"
        f.locale = Locale(identifier: "en_US_POSIX")
        f.timeZone = TimeZone.current
        return f
    }()
}

extension JSONEncoder {
    static var sessionEncoder: JSONEncoder {
        let encoder = JSONEncoder()
        encoder.dateEncodingStrategy = .formatted(Session.dateFormatter)
        encoder.outputFormatting = .prettyPrinted
        return encoder
    }
}

extension JSONDecoder {
    static var sessionDecoder: JSONDecoder {
        let decoder = JSONDecoder()
        decoder.dateDecodingStrategy = .formatted(Session.dateFormatter)
        return decoder
    }
}
